### DISCUSSION TITLE
Accessing a field name which is an alias

### DISCUSSION THREAD

**Discussion by falstaff1288** [cue-lang/cue#1555] [Q&A] [answered]
Hello,

I'm trying to define & call multiple different structs at once using a unique name. The idea is as shown below:

```
package test

#AppTemplates: {
    nginx: [ID=_]: {
        application: {
            "\(ID)": {
                kind: "Application"
            }
        }
        secret: {
            "\(ID)": {
                kind: "Secret"
            }
        }
    }
}

// Calling the template
#AppTemplates.nginx.nginx1
```

Basically, the alias` ID` would be replaced by `nginx1`. This the output I'm getting:
```
$ cue export package_apps.cue --out yaml
undefined field: nginx1:
    ./package_apps.cue:18:21
```

Expected result
```
$ cue export package_apps.cue --out yaml
application:
  nginx1:
    kind: Application
secret:
  nginx1:
    kind: Secret
```

Any ideas if there's a way to achieve this?

**Comment by verdverm** [comment, marked as answer]
You need another line like `#AppTemplates: nginx: nginx1: _` to define the named instances. In your example, you have only defines the pattern and no actual instances. This is why trying to reference it fails.

Note, I would call the last line "referencing" and not "calling" and that when you run `cue`, you render or export the configuration.

**Reply by falstaff1288** [reply]
I see. Makes sense.

Noted for the correct terminology. I'm slowing getting used to it.

Thanks!
